### ISSUE TITLE
Include the issue ID and whether the change could be fixed in the report file

### ISSUE THREAD

**Issue by castholm** [dotnet/format#1057]
I'm trying write some automation scripts to make formatting code using dotnet-format part of our build pipeline. If dotnet-format is unable to apply fixes (thus exiting with a non-zero exit code) I'd like to use the generated report file to log issues and abort the build.

Example PowerShell script illustrating what I am trying to do:
```powershell
dotnet format --report ".\report.json"

if ($LASTEXITCODE -ne 0) {
    # There were unfixable issues.
    $json = Get-Content ".\report.json" | Out-String | ConvertFrom-Json
    $issues = $json |
        Select-Object FilePath -ExpandProperty FileChanges |
        Select-Object FilePath, LineNumber, CharNumber, FormatDescription

    # Log issues to the console.
    Write-Host "[ERROR] Unable to automatically fix $( $issues.Count ) issues:" -ForegroundColor Red
    $issues |
        %{ "[ERROR] $( $_.FilePath )($( $_.LineNumber ),$( $_.CharNumber )): error : $( $_.FormatDescription )" } |
        Write-Host -ForegroundColor Red

    exit 1
}

# Proceed with the build script as normal...

```

One problem I've encountered is that since the report file includes both fixed and unfixable issues, there is no way to programmatically tell which issues are unfixable. If the report included some sort of boolean property indicating whether a fix was applied, it would make it possible to isolate fixed issues from unfixable issues.

Another inconvenience is that the report file does not include the issue ID as a property. You can generally extract the issue ID from `FormatDescription` using regex, but it would be more convenient if it had its own property instead of being embedded in the message.

To summarize, my suggestion is to extend `FileChange` with the following properties:

```diff
 public class FileChange
 {
     public int LineNumber { get; }

     public int CharNumber { get; }
+
+    public string? IssueId { get; }

     public string FormatDescription { get; }
+
+    public boolean Fixed { get; }
 }
```

**Comment by castholm** [comment]
>For 1. and 2. and 3. we could do as you suggest.
>For 4. and 5. I assume you would also want some additional error info?

For the use case I'm trying to support (alerting the host running the script and aborting), I mainly just want to report and highlight issues that can't be resolved automatically and require manual human intervention. Whether this is because no fix exists or because the analyzer crashed doesn't matter; if it happens to be because of a crash I'm thinking you could still always check the lines logged by dotnet-format.

That said, more detailed error information included in the report could always be useful, but it's not necessary for this use case.
